src/projections/spilhaus.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Implementation of the Spilhaus projections based on Adams World in a Square
+ * II.
+ *
+ * Explained in https://github.com/OSGeo/PROJ/issues/1851
+ *
+ * Copyright (c) 2025 Javier Jimenez Shaw
+ *
+ * Related material
+ * ----------------
+ *
+ * Map Projections - A Working Manual. 1987. John P. Snyder
+ * Sections 3 and 5.
+ * https://doi.org/10.3133/pp1395
+ * Online at https://neacsu.net/docs/geodesy/snyder/2-general/
+ */
+
+#include <errno.h>
+#include <math.h>
+
+#include "proj.h"
+#include "proj_internal.h"
+
+C_NAMESPACE PJ *pj_adams_ws2(PJ *);
+
+PROJ_HEAD(spilhaus, "Spilhaus") "\n\tSph&Ell";
+
+namespace { // anonymous namespace
+struct pj_spilhaus_data {
+    double cosalpha;
+    double sinalpha;
+    double beta;
+    double lambda_0;
+    double conformal_distortion;
+    double cosrot;
+    double sinrot;
+    PJ *adams_ws2;
+};
+
+} // anonymous namespace
+
+static PJ_XY spilhaus_s_forward(PJ_LP lp, PJ *P) {
```

Comment:
the s_ means spherical, so:

```suggestion
static PJ_XY spilhaus_forward(PJ_LP lp, PJ *P) {
```